### ISSUE TITLE
Client does not check for presence of system tray at startup

### ISSUE THREAD

**Issue by Deaddy** [owncloud/client#10996]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

The client does not check whether or not a system tray is present.

On startups beyond the first startup (where one is prompted for configuration) nothing visible happens, i.e. one only sees
```
$ Downloads/ownCloud-4.1.0.11250-x86_64.AppImage
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-deaddy'
m_crashReporterChar:  /tmp/.mount_ownCloirZSjT/usr/bin/owncloud_crash_reporter
```
but that's it.

If the client is already configured and still authorized, actually synchronization seems to work.

Further invocations of the command only say that the GUI application already is running:
```
Downloads/ownCloud-4.1.0.11250-x86_64.AppImage
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-deaddy'
gui.application: Already running, exiting...
```

### Expected behavior

I expect to see the client.

In older versions invocation of the owncloud command even when the client was already running would also foreground the application, what was nice.

One simply has to check `QSystemTrayIcon::isSystemTrayAvailable()` and only hide/minimze the client if this evaluates to true.

#### Workaround:
Start client with `-s` parameter.

### Steps to reproduce the issue

1. not have a traybar
2. start an already configured client

### Screenshots

nothing is visible, that is the point. so no screenshot :-)

### Logs

See above.

### Client version number

ownCloud-4.1.0.11250-x86_64.AppImage 279a6f
Qt 5.15.8

### Desktop environment (Linux only)

anything that is not a desktop environment (e.g. xmonad, lab computers who only have rudimentary x, servers) or where no tray bar is visible

### Server information

Independent of server versions (multiple oc10.12.x and ocis-3.x versions) tested as target

### Additional context

Qt detection of system tray seems to work as expected, as this [stolen](https://stackoverflow.com/questions/72687141/how-to-check-system-tray-availability-programmatically) code easily demonstrates:

is_tray_available.cpp:
```
#include <QApplication>
#include <QSystemTrayIcon>

int main(int argc, char** argv)
{
  QApplication app(argc, argv);
  if(QSystemTrayIcon::isSystemTrayAvailable()) {
    printf("yes, tray is available\n");
  } else {
    printf("no, tray is not available\n");
  }

  return 0;
}
```
is_tray_available.pro:
```
QT += widgets
TARGET = is_tray_available
SOURCES += is_tray_available.cpp
```
and then build with `qmake && make` to yield
```
$ ./is_tray_available 
no, tray is not available
```

**Comment by fmoc** [comment]
Our docs suggest you need a system tray, as this is an essential part of the user interface. The client is supposed to run in the background. See https://doc.owncloud.com/desktop/4.0/installing.html#installation-on-linux.

If you do not have a system tray for some reason, you can always launch the client with `-s` resp. `--showsettings`. This will always bring the main user interface to the front.

**Comment by Deaddy** [comment]
Yes, as written I am aware of that workaround.

The docs admitting that major distributions do not have a tray anymore just increases the relevance of this regression, especially since the check for the sys tray takes about two lines and the corresponding override mechanism is already in place and just needs to be exapted.

From our supporter's view, it is also a very hard issue to debug, because when lab teams or similar report "client not working anymore after update", there is very little on client side the dreaded end user can procure in terms of information and it takes a long ticket ping pong match until the issue can be resolved, leading to frustration on all sides.

**Comment by TheOneRing** [comment]
The client is supposed to run in the background, if we would always display the main window if we have no systray it would be annoying to quite a few users.
We however install a owncloud.desktop file which calls the client with -s so if you use your launcher application and click the owncloud entry the client is supposed to come to the front.
Additionally the entry should provide you with two context menu actions, "Show" the client and "Quit".

That way the client is supposed to only show a ui, if a user intends to interact with it.
Another way to see the status of the client, is to use a shell extension, which will display the sync state of your files.
If you would like to see the client status in a decent way I'd recommend to either install a systray or switch to a sane desktop environment.

**Comment by fmoc** [comment]
> regression

This is not a regression.

TL;DR: As @TheOneRing pointed out, by default, when launched from any kind of app launcher menu, `-s` is already provided. When using the CLI, one needs to pass `-s` manually.
